File: dotnet/Varar.Core/Parse.cs
namespace Varar.Core;

public static class Parse
{
    /// <summary>Parses Markdown source into a <see cref="Doc"/>. Port of <c>parse.ts</c>.</summary>
    public static Doc Run(string path, string source) =>
        Structurer.Structure(path, source, Scanner.Scan(source));
}

File: rust/core/src/parse.rs
//! Top-level parse entry point: `scan` then `structure` — port of `parse.ts` /
//! `Parse.java`.

use crate::ast::Doc;
use crate::{scanner, structurer};

/// Parses `source` into a [`Doc`].
pub fn parse(path: &str, source: &str) -> Doc {
    structurer::structure(path, source, scanner::scan(source))
}
